src/store/Controller.cc

```diff
@@ -443,6 +443,12 @@ Store::Controller::transientsWriter(const StoreEntry &e) const
     return transients && e.hasTransients() && transients->isWriter(e);
 }
 
+bool
+Store::Controller::cacheWriter(const StoreEntry &e) const
+{
+    return transients ? transientsWriter(e) : true;
```

Comment:
Consider simplifying(?):
```suggestion
    return !transients || transientsWriter(e);
```

I do not insist on this change. Your call.